libs/src/utils/multiProvider.js

```diff
@@ -0,0 +1,44 @@
+const { promisify, callbackify } = require('util')
+const Web3 = require('../web3')
+const { shuffle } = require('lodash')
+
+class MultiProvider extends Web3.providers.HttpProvider {
+  // web3 consumes a provider object on initialization
+  // ref: https://github.com/ChainSafe/web3.js/blob/1.x/packages/web3/types/index.d.ts#L31
+  // which references: https://github.com/ChainSafe/web3.js/blob/1.x/packages/web3-core/types/index.d.ts#L436
+  // MultiProivder implements AbstractProvider which can be consumed by web3
+
+  constructor (providers) {
+    super()
```

Comment:
how does this work to call super constructor with no args? don't you need to pass a url string in?
per https://github.com/ChainSafe/web3.js/blob/1.x/packages/web3/types/index.d.ts#L31

prob for consistency we should set the default provider

Reply:
https://github.com/ChainSafe/web3.js/blob/1.x/packages/web3-providers-http/types/index.d.ts#L55 shows that all inputs to `HttpProvider` are optional.

I don't think adding a default provider here would help. Since we replace the send function, which is the only method that uses the input to `HttpProvider` and the other functions are no-op functions.

I am not opposed to set the first provider here in the initialization if you have further thoughts on this, thanks

Reply:
yeah i see its optional but i don't see the benefit to not keep functionality changes as minimal as possible. previously we would construct the web3 instance with a providers param so we should keep it that way. even if it seems that send is the only function that uses it, don't see why we would introduce more potential for error when its simple to keep as is?